### PR TITLE
TIMRLINKN-60 Can't import the same tasks csv multiple times

### PR DESCRIPTION
**Pull request by gugmaster**

### CODE REVIEW

File: timrlink.net.Core/Service/TaskService.cs [hunk 1]
```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
         {
             logger.LogDebug($"Checking Task(Name={task.name}, ExternalId={task.externalId})");
 
-            if (existingTaskIDs.TryGetValue(task.externalId, out var existingTask) && updateTask)
+            existingTaskIDs.TryGetValue(task.externalId, out var existingTask);
```
Comment (daberni) [inline timrlink.net.Core/Service/TaskService.cs:160]: why are you no longer combining this with the if-statement?

File: timrlink.net.Core/Service/TaskService.cs [hunk 2]
```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
...
+            
+            if (addedTasks.TryGetValue(task.externalId, out var newAddedTask))
```
Comment (daberni) [inline timrlink.net.Core/Service/TaskService.cs:162]: whats the reason of changing the order?

File: timrlink.net.Core/Service/TaskService.cs [hunk 3]
```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
...
-                    await timrSync.UpdateTaskAsync(new API.UpdateTaskRequest(task)).ConfigureAwait(false);
-                    existingTaskIDs[task.externalId] = task;
-                }    
+                    if (updateTask) {
+                        logger.LogInformation(
+                            $"Updating existing Task(Name={task.name}, ExternalId={task.externalId})");
+                        await timrSync.UpdateTaskAsync(new API.UpdateTaskRequest(task)).ConfigureAwait(false);
+                        existingTaskIDs[task.externalId] = task;
+                    }
+                    else
```
Comment (daberni) [inline timrlink.net.Core/Service/TaskService.cs:182]: probably the core change just resolves around adding an else block here right? what are the remaining changes for?

**Review by daberni — CHANGES_REQUESTED** [review]
Any testcases?

File: timrlink.net.Core/Service/TaskService.cs (hunk 3 above)
Comment (gugmaster) [reply timrlink.net.Core/Service/TaskService.cs:182]: Sure. So you like the old style more we had? I thought it's a bit more readable this way. But anyway.

Comment (daberni) [reply timrlink.net.Core/Service/TaskService.cs:182]: I don't understand. Which particular part do you like better? The else here is required for sure.

But why did you use `existingTask != null`? If you already have the boolean of `.TryGetValue()` which is a standard pattern in C#

File: timrlink.net.Core/Service/TaskService.cs [hunk 4]
```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
...
+                    {
+                        logger.LogWarning(
+                            $"Task(Name={task.name}, ExternalId={task.externalId}) already exists but with different properties - skipping");
```
Comment (daberni) [inline timrlink.net.Core/Service/TaskService.cs:185]: `Skip updating modified Task()` would be more consistent with other messages

File: timrlink.net.Core/Service/TaskService.cs [hunk 5]
```diff
@@ -157,23 +157,38 @@ protected async Task AddOrUpdateTask(IDictionary<string, API.Task> existingTaskI
...
+                    }
+                }
+                else
+                {
+                    logger.LogInformation($"Task(Name={task.name}, ExternalId={task.externalId}) already exits with the same properties");
```
Comment (daberni) [inline timrlink.net.Core/Service/TaskService.cs:190]: `Skip updating unmodified Task()` would be more consistent with other messages

**Pull request merged**